src/main/java/com/thebois/models/beings/AbstractBeing.java

```diff
@@ -153,8 +160,40 @@ protected void setPath(final Collection<Position> path) {
         return Optional.of(path.peek().deepClone());
     }
 
+    protected Position getFinalDestination() {
+        if (path.isEmpty()) return null;
+        return path.firstElement().deepClone();
+    }
+
     protected IPathFinder getPathFinder() {
         return pathFinder;
     }
 
+    protected IFinder getFinder() {
+        return this.finder;
+    }
```

Comment:
I think it's because fields should always be private, and they should be accessed through getters instead. As a subtype is a kind of client (albeit a more closely related one), you'd want the subtype to go through the getters as well.

Reply:
On a related note (so I don't have to create a new review):

@JacobBredin you should probably rename this `IFinder` to be much more specific, since I've currently got a separate `IResourceFinder` on my branch. Maybe `IStructureFinder`.

Reply:
Done!